hash_test.go

```diff
@@ -39,6 +39,27 @@ func cryptoToHash(h crypto.Hash) func() hash.Hash {
 	return nil
 }
 
+func TestHashNotMarshalable(t *testing.T) {
+	h := openssl.NewSHA256()
+	state, err := h.(encoding.BinaryMarshaler).MarshalBinary()
+	if err != nil {
+		t.Fatal(err)
```

Comment:
Maybe it's worth mentioning in the error or a comment that this test assumes that SHA256 is always marshallable due to what we support in 1.23? (Or rather, what we don't support. 🙂)

Reply:
Good point, done!